### DISCUSSION TITLE
Remove "Click" Terminology and Replace with "Press"

### DISCUSSION THREAD

**Discussion by robloo** [AvaloniaUI/Avalonia#7421] [General]
I know this will be frowned upon by many but hear me out:

"Click" is an old term based on the sound a computer mouse makes when pressed by the user. Modern systems have many other types of input besides mouse: such as touch. For input systems like touch "Click" doesn't make sense and is a remnant of out-of-date thinking. It exists today primary for compatibility and because it was always so.

My proposal would be to remove this term from the entire Avalonia framework and replace it as follows:

 *  "Click" -> "Press"
 *  "Clicked" -> "Pressed"

This would affect events and events args for controls such as `Button`. It would modernize the framework and make it ready for touch-first systems like mobile.

An alternative term is "Tapped" but that should apply ONLY for touch-based events as is done already. General-purpose events that can be invoked from pointers, touch, etc. should use the 'Pressed" terminology.

This was partially done already for UWP/WinUI where several events now use the Pressed terminology. However, for backwards compatibly it still uses Click in many places as well.

**Comment by kekekeks** [comment]
>An alternative term is "Tapped" but that should apply ONLY for touch-based events as is done already. 

Actually Tapped event is raised for all input types [just like with UWP](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.tapped?view=winrt-22000#remarks).

**Reply by robloo** [reply]
Yes, my comment above is incorrect. My thinking was more that the Tapped, DoubleTapped, etc. events were added in metro apps when touch support was first implemented. Those events were a direct result of touch/gesture support as a first-class input method and named accordingly. Of course for app compatibility those same events need to be fired for other input devices as well.
